test/e2e/storage/non_graceful_node_shutdown.go

```diff
@@ -100,8 +100,20 @@ var _ = utils.SIGDescribe("[Feature:NodeOutOfServiceVolumeDetach] [Disruptive] [
 			pod := createAndVerifyStatefulDeployment(ctx, scName, deploymentName, ns, podLabels, c)
 			oldNodeName := pod.Spec.NodeName
 
+			cleanupItems := []func(){}
+			cleanup := func() {
+				for _, c := range cleanupItems {
+					c()
+				}
+				cleanupItems = nil
+			}
+			defer cleanup()
+
 			ginkgo.By("Stopping the kubelet non gracefully for pod" + pod.Name)
 			utils.KubeletCommand(ctx, utils.KStop, c, pod)
+			cleanupItems = append(cleanupItems, func() {
+				utils.KubeletCommand(ctx, utils.KStart, c, pod)
```

Comment:
you can use `ginkgo.DeferCleanup` directly and avoid the `cleanup func()`
cc: @pohly

Reply:
This is using the `ctx` from the test, which is wrong: after a failure, that context is cancelled and cannot be used anymore. A simpler and correct invocation is:

```suggestion
				ginkgo.DeferCleanup(utils.KubeletCommand, utils.KStart, c, pod)
```

Reply:
Done